snapcraft/tests/commands/test_snap.py

```diff
@@ -29,6 +31,100 @@
 
 from snapcraft.main import main
 from snapcraft import tests
+from snapcraft.tests.test_lxd import check_output_side_effect
+
+
+class SnapCommandContainerized(tests.TestCase):
+
+    yaml_template = """name: snap-test
+version: 1.0
+summary: test containerized snap
+description: if snap is succesful a snap package will be available
+architectures: ['amd64']
+type: {}
+confinement: strict
+grade: stable
+
+parts:
+    part1:
+      plugin: nil
+"""
+
+    def setUp(self):
```

Comment:
You should create a common `ContainerTestClass` with this common `setUp` (and `make_snapcraft_yaml`) that both Cleanbuild and this class can inherit from. If the only thing that really makes these different is the arguments to `main` then please consider using `testscenarios`.